config/v1/types_infrastructure.go

```diff
@@ -478,6 +478,13 @@ type AWSPlatformStatus struct {
 	// +kubebuilder:validation:MaxItems=25
 	// +optional
 	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+
+	// dnsConfig contains information about the type of DNS solution in use
+	// for the cluster and its related configuration.
+	// +default={"provider": "ClusterProvided"}
+	// +kubebuilder:default={"provider": "ClusterProvided"}
+	// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+	DNSConfig *DNSConfigurationType `json:"dnsConfig"`
```

Comment:
Kubebuilder will automatically add the default value of provider="ClusterProvided". There is a integration test to that effect.

Reply:
> Kubebuilder will automatically add the default value of provider="ClusterProvided". There is a integration test to that effect.

Interesting. I didn't know you could do that! That's cool. 

I'm having a hard time wrapping my head around the idea of something being both _required_ and having a _default_ value; to me they seem mutually exclusive, or at least to mutually diminish one another. Just a personal observation/opinion.

Joel helpfullly brought up UPI installs--we don't technically know that all existing clusters are using IPI-style cloud DNS. Users could opt for UPI in order to establish their own non-route53 DNS solutions. I don't know how many that would be, but I'm pretty confident it's non-zero. So if we default this value, I think the values would likely be incorrect in some cases.

Reply:
With the current implementation, if DNSConfig.Provider is left unset by the user, it will be given the default of "CloudProvided". No further action will be taken within OpenShift when Provider is "CloudProvided". Afaik, this should work for UPI too. Are there some functionality/semantics issues that I am missing?

Reply:
I replaced "CloudProvided" with "Default" to indicate that that the right defaults for UPI or IPI are in use.

Reply:
Replaced "Default" with "".